x-pack/plugins/security_solution/public/common/components/exceptions/use_fetch_alert_data.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useState } from 'react';
+import { Ecs } from '../../../../common/ecs';
+import { fetchQueryAlerts } from '../../../detections/containers/detection_engine/alerts/api';
+import { AlertSearchResponse } from '../../../detections/containers/detection_engine/alerts/types';
+
+import { buildGetAlertByIdQuery } from './helpers';
+import { FlattenType } from './types';
+
+type Func = () => Promise<void>;
+
+interface ReturnQueryAlerts {
+  loading: boolean;
+  data: AlertSearchResponse<EcsHit> | null;
+  response: string;
+  request: string;
+  refetch: Func | null;
+}
+
+interface EcsHit {
+  _id: string;
+  _index: string;
+  _source: {
+    '@timestamp': string;
+  } & Omit<FlattenType<Ecs>, '_id' | '_index'>;
+}
+
+export type Alert = {
+  '@timestamp': string;
+} & FlattenType<Ecs>;
+
+export const useFetchAlertData = (
```

Comment:
Why can't we use the existing `useQueryAlerts` hook here?

Reply:
The reason i switched was because this we were using the `useQueryAlerts` hook conditionally and typescript doesn't like it. Moving the conditional logic within the hook itself fixes that

Reply:
Switched to using outer wrapper